File: hw/paillacho/U3/HW20Composite/Composite/src/main/java/ec/edu/espe/composite/model/Manager.java
package ec.edu.espe.composite.model;

/**
 *
 * @author Carlos Paillacho
 */
public class Manager extends Supervisor {

    /**
     * Default constructor sets the title.
     */
    public Manager() {
        title = "Manager";
    }

    /**
     * Constructs a Manager with a name.
     *
     * @param aName the manager's name
     */
    public Manager(String aName) {
        this();
        name = aName;
    }

    /**
     * Custom processing for manager naming could go here.
     */
    @Override
    public void stateName() {
        // Additional behavior for Manager if needed
        super.stateName();
    }
}

File: hw/bonilla/U3/HW20Composite/Compositejs/model/Manager.js
const Supervisor = require('./Supervisor');

class Manager extends Supervisor {
  constructor(aName) {
    super();
    this.title = "Manager";
    if (aName) this.name = aName;
  }

  stateName() {
    super.stateName();
  }
}

module.exports = Manager;
